samples/nrf9160/gps/src/main.c

```diff
@@ -37,44 +53,59 @@ static bool           update_terminal;
 static u64_t          fix_timestamp;
 nrf_gnss_data_frame_t last_fix;
 
+K_SEM_DEFINE(lte_ready, 0, 1);
+
 void bsd_recoverable_error_handler(uint32_t error)
 {
 	printf("Err: %lu\n", (unsigned long)error);
 }
 
-static int enable_gps(void)
+static int setup_modem(void)
 {
-	int  at_sock;
-	int  bytes_sent;
-	int  bytes_received;
-	char buf[2];
+	for (int i = 0; i < ARRAY_SIZE(at_commands); i++) {
 
-	at_sock = socket(AF_LTE, 0, NPROTO_AT);
-	if (at_sock < 0) {
-		return -1;
+		if (at_cmd_write(at_commands[i], NULL, 0, NULL) != 0) {
+			return -1;
+		}
 	}
 
-	for (int i = 0; i < ARRAY_SIZE(at_commands); i++) {
-		bytes_sent = send(at_sock, at_commands[i],
-				  strlen(at_commands[i]), 0);
+	return 0;
+}
 
-		if (bytes_sent < 0) {
-			close(at_sock);
+static void wait_for_lte(char *response)
+{
+	if (!memcmp(status1, response, AT_CMD_SIZE(status1)) ||
+		!memcmp(status2, response, AT_CMD_SIZE(status2)) ||
+		!memcmp(status3, response, AT_CMD_SIZE(status3)) ||
+		!memcmp(status4, response, AT_CMD_SIZE(status4))) {
+		k_sem_give(&lte_ready);
+	}
+}
+
+static int activate_lte(bool activate)
+{
+	if (activate) {
+		if (at_cmd_write(AT_ACTIVATE_LTE, NULL, 0, NULL) != 0) {
 			return -1;
 		}
 
-		do {
-			bytes_received = recv(at_sock, buf, 2, 0);
-		} while (bytes_received == 0);
+		at_cmd_set_notification_handler(wait_for_lte);
```

Comment:
You should use the `at_notif` API to register a callback.

Reply:
Ok, was not aware that this was done.

Reply:
Code updated